tests/test_git_init.py

```diff
@@ -1,29 +1,21 @@
 """Checks that the git repo initialisation works."""
 
-import pathlib
 import subprocess
 import typing
 
 import pytest
 
 
-@pytest.mark.parametrize("initialise_git_repository", [True, False])
+@pytest.mark.parametrize("init", [True, False])
 def test_initialisation_of_git_repo(
-    initialise_git_repository: bool,  # noqa: FBT001
+    init: bool,  # noqa: FBT001
+    default_config_with: typing.Callable,
     generate_package: typing.Callable,
-    tmp_path: pathlib.Path,
 ) -> None:
     """Checks to see if git was correctly initialised if desired."""
-    test_config = {
-        "github_owner": "test-user",
-        "project_short_description": "description",
-        "project_name": "Cookiecutter Test",
-        "initialise_git_repository": initialise_git_repository,
-    }
     # Run cookiecutter with initialise_git_repository
-    result = generate_package(config=test_config, path=tmp_path)
-
-    test_project_dir = tmp_path / "cookiecutter-test"
+    config = default_config_with("initialise_git_repository", str(init))
```

Comment:
```suggestion
    config = default_config_with(initialise_git_repository=str(init))
```

If we go with suggested generalisation of `default_config_with` to use `**kwargs`